execution_engine/src/core/engine_state/mod.rs

```diff
@@ -679,20 +709,50 @@ where
             }
         };
 
-        let proposer_main_purse_balance_key = {
-            let proposer_main_purse = proposer_account.main_purse();
+        let rewards_target_purse = {
+            let proposer_account: Account = match tracking_copy
+                .borrow_mut()
+                .get_account(correlation_id, AccountHash::from(&proposer))
+            {
+                Ok(account) => account,
+                Err(error) => {
+                    return Ok(ExecutionResult::precondition_failure(error.into()));
+                }
+            };
+
+            match self.config.fee_handling() {
+                FeeHandling::PayToProposer { .. } => {
+                    // the proposer of the block this deploy is in receives the gas from this deploy
+                    // execution
+                    proposer_account.main_purse()
+                }
+                FeeHandling::Accumulate => {
+                    let mint_hash = self.get_system_mint_hash(correlation_id, prestate_hash)?;
+
+                    let mint_contract = tracking_copy
+                        .borrow_mut()
+                        .get_contract(correlation_id, mint_hash)?;
+
+                    let rewards_purse_uref = mint_contract.named_keys().get(REWARDS_PURSE_KEY).and_then(|key| (*key).into_uref()).unwrap_or_else(|| {
+                        error!("fee elimination is configured to accumulate but mint does not have rewards purse; defaulting to a proposer");
+                        proposer_account.main_purse()
```

Comment:
Fair point, lack of an accumulation purse in the handle payment purse is a configuration problem and is treated like one now - it could happen when you switch `fee_handling` mode.